cmd/nodecmd/create.go

```diff
@@ -217,40 +216,88 @@ func createNodes(_ *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-
-	time.Sleep(30 * time.Second)
-
+	if err := updateAnsiblePublicIPs(clusterName); err != nil {
+		return err
+	}
+	inventoryPath := app.GetAnsibleInventoryDirPath(clusterName)
 	avalancheGoVersion, err := getAvalancheGoVersion()
 	if err != nil {
 		return err
 	}
-	inventoryPath := app.GetAnsibleInventoryDirPath(clusterName)
 	if err = ansible.CreateAnsibleHostInventory(inventoryPath, cloudConfig.CertFilePath, cloudService, publicIPMap); err != nil {
 		return err
 	}
+	allHosts, err := ansible.GetInventoryFromAnsibleInventoryFile(inventoryPath)
+	if err != nil {
+		return err
+	}
+	hosts := ansible.FilterHostsByNodeID(allHosts, cloudConfig.InstanceIDs)
+	// waiting for all nodes to become accessible
+	failedHostMap := waitForHosts(hosts)
+	failedHosts := []string{}
+	for node, err := range failedHostMap {
+		ux.Logger.PrintToUser("Instance %s failed to provision with error %s. Please check instance logs for more information", node, err)
+		failedHosts = append(failedHosts, node)
+	}
+	if len(failedHosts) > 0 {
+		return fmt.Errorf("failed to provision node(s) %s", failedHosts)
+	}
 
-	ux.Logger.PrintToUser("Installing AvalancheGo and Avalanche-CLI and starting bootstrap process on the newly created Avalanche node(s) ...")
 	ansibleHostIDs, err := utils.MapWithError(cloudConfig.InstanceIDs, func(s string) (string, error) { return models.HostCloudIDToAnsibleID(cloudService, s) })
 	if err != nil {
 		return err
 	}
-	createdAnsibleHostIDs := strings.Join(ansibleHostIDs, ",")
-	if err = runAnsible(inventoryPath, network, avalancheGoVersion, clusterName, createdAnsibleHostIDs); err != nil {
-		return err
+	ux.Logger.PrintToUser("Installing AvalancheGo and Avalanche-CLI and starting bootstrap process on the newly created Avalanche node(s) ...")
+	nodeResultChannel := make(chan models.NodeErrorResult, len(hosts))
+	parallelWaitGroup := sync.WaitGroup{}
+	for _, host := range hosts {
+		parallelWaitGroup.Add(1)
+		go func(nodeResultChannel chan models.NodeErrorResult, host models.Host) {
+			defer parallelWaitGroup.Done()
+			if err := provideStakingCertAndKey(host); err != nil {
```

Comment:
all of this to use the same ssh connection on host. end with a new host.CloseConn function call

Reply:
goph doesn't let me reuse the connection unfortunatelly